qcelemental/models/procedures.py

```diff
@@ -131,6 +133,73 @@ def _trajectory_protocol(cls, v, values):
         return v
 
 
+class OptimizationSpecification(ProtoModel):
+    """
+    A specification for how a geometry optimization should be performed inside of another procedure.
+    """
+
+    schema_name: constr(strip_whitespace=True, regex="qcschema_optimization_specification") = "qcschema_optimization_specification"  # type: ignore
+    schema_version: int = 1
+
+    procedure: str = Field(..., description="Optimization procedure to run the optimization with.")
+    keywords: Dict[str, Any] = Field({}, description="The optimization specific keywords to be used.")
+    protocols: OptimizationProtocols = Field(OptimizationProtocols(), description=str(OptimizationProtocols.__doc__))
+
+    @validator("procedure")
+    def _check_procedure(cls, v):
+        return v.lower()
+
+
+class TorsionDriveInput(ProtoModel):
+
+    schema_name: constr(strip_whitespace=True, regex=qcschema_torsion_drive_input_default) = qcschema_torsion_drive_input_default  # type: ignore
+    schema_version: int = 1
+
+    keywords: Dict[str, Any] = Field({}, description="The torsion drive specific keywords to be used.")
```

Comment:
I wonder if this [TDKeyword model ](https://github.com/MolSSI/QCFractal/blob/5a8e4dc309ac9a7a7b2cf023b03279aca0600e2c/qcfractal/interface/models/torsiondrive.py#L21) should be used here to ensure correct validation, moving this to qcelemental would mean it could be used in QCFractal still making the two consistent.

Reply:
@bennybp, is moving models from qcf to qcel encouraged, discouraged, impossible? shifting it here sounds like a sensible move to me, but I may be naive.

Reply:
@bennybp are you opposed to migration of `TDKeywords` from QCFractal to QCElemental?

I don't see any dependence of `TDKeywords` on any Fractal components, so should be straightforward. I'm for doing it.

Reply:
I'll add `TDKeywords` to this PR and merge to unblock. I think @bennybp and I can adapt in QCFractal to this.

Reply:
Ah, just realized I can't add `TDKeywords` myself. @SimonBoothroyd can you go ahead and pull the `TDKeyword` model out of `QCFractal` and add it to this PR, using it for validation as @jthorton suggested?

Talking to @bennybp, no immediate objections. He's currently oversubscribed, but having this included will ensure it's here when he performs his review.

Reply:
Sounds good - added in 63e02b7. The only difference is that is doesn't call the recursive normaliser on initialisation as this is not exposed in QCE. @dotsdl you should also now have colab access to my fork.

Reply:
Awesome, thank you @SimonBoothroyd!

@bennybp is there anything additional we can add to assist in review?